client/pd_service_discovery.go

```diff
@@ -158,34 +156,15 @@ type pdServiceClient struct {
 	networkFailure atomic.Bool
 }
 
-func newPDServiceClient(addr, leaderAddr string, tlsCfg *tls.Config, conn *grpc.ClientConn, isLeader bool) ServiceClient {
-	var httpAddress string
-	if tlsCfg == nil {
-		if strings.HasPrefix(addr, httpsScheme) {
-			addr = strings.TrimPrefix(addr, httpsScheme)
-			httpAddress = fmt.Sprintf("%s%s", httpScheme, addr)
-		} else if strings.HasPrefix(addr, httpScheme) {
-			httpAddress = addr
-		} else {
-			httpAddress = fmt.Sprintf("%s://%s", httpScheme, addr)
-		}
-	} else {
-		if strings.HasPrefix(addr, httpsScheme) {
-			httpAddress = addr
-		} else if strings.HasPrefix(addr, httpScheme) {
-			addr = strings.TrimPrefix(addr, httpScheme)
-			httpAddress = fmt.Sprintf("%s%s", httpsScheme, addr)
-		} else {
-			httpAddress = fmt.Sprintf("%s://%s", httpsScheme, addr)
-		}
-	}
-
+// NOTE: In the current implementation, the address passed in is bound to have an http scheme,
+// because it is processed in `newPDServiceDiscovery`, and the url returned by etcd member is its own.
+// When testing, the address is also bound to have an http scheme.
+func newPDServiceClient(addr, leaderAddr string, conn *grpc.ClientConn, isLeader bool) ServiceClient {
 	cli := &pdServiceClient{
```

Comment:
Do we still need `GetHTTPAddress` function?
[in line ](https://github.com/tikv/pd/pull/7901/files#diff-76173cab783101dc7dfff41befacf2d2a1339ea83eb57ca2b88e6508180cd149R184)